lib/xebow_web/live/matrix_live.ex

```diff
@@ -0,0 +1,191 @@
+defmodule XebowWeb.MatrixLive do
+  use XebowWeb, :live_view
+
+  alias RGBMatrix.{Animation, Engine}
+
+  @layout Xebow.layout()
+  @black Chameleon.HSV.new(0, 0, 0)
+  @black_frame @layout
+               |> Layout.leds()
+               |> Map.new(fn led ->
+                 {led.id, @black}
+               end)
+
+  @impl Phoenix.LiveView
+  def mount(_params, _session, socket) do
+    if connected?(socket) do
+      register_with_engine!()
+    end
+
+    {config, config_schema} = Engine.get_animation_config()
+
+    initial_assigns = [
+      leds: make_view_leds(@black_frame),
+      config: config,
+      config_schema: config_schema,
+      animation_types: Animation.types(),
+      current_animation_index: 0
+    ]
+
+    {:ok, assign(socket, initial_assigns)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info({:render, frame}, socket) do
+    {:noreply, assign(socket, leds: make_view_leds(frame))}
```

Comment:
I've found this pattern of updating the assigns directly in the LiveView can get unwieldy as the project gets larger. A new pattern I'm following that I like more is to create a "model" module which manages the state of the live view and updating with `Phoenix.LiveView.update/3`. 

You can see an example in this blog post: https://hostiledeveloper.com/2020/06/19/organizing-liveview-logic-with-presentation-models.html

We can refactor this later, just something to keep in mind!

Reply:
@nickdichev that's a cool example in the article! I like how it improves the readability of the `handle_*` functions as well!

Reply:
https://github.com/ElixirSeattle/xebow/issues/71